ami/trainers/i_jepa_latent_visualization_trainer.py

```diff
@@ -0,0 +1,217 @@
+from ami.models.model_wrapper import ModelWrapper
+from ami.tensorboard_loggers import StepIntervalLogger
+
+from .base_trainer import BaseTrainer
+
+
+class IJEPALatentVisualizationTrainer(BaseTrainer):
+    """Trainer for IJEPA Decoder."""
+
+    @override
+    def __init__(
+        self,
+        partial_dataloader: partial[DataLoader[Tensor]],
+        partial_optimizer: partial[Optimizer],
+        device: torch.device,
+        logger: StepIntervalLogger,
+        encoder_name: Literal[ModelNames.I_JEPA_CONTEXT_ENCODER, ModelNames.I_JEPA_TARGET_ENCODER],
+        decoder_name: Literal[ModelNames.I_JEPA_CONTEXT_DECODER, ModelNames.I_JEPA_TARGET_DECODER],
+        max_epochs: int = 1,
+        minimum_dataset_size: int = 1,
+        minimum_new_data_count: int = 0,
+        num_visualize_images: int = 64,
+        visualize_grid_row: int = 8,
+    ) -> None:
+        """Initializes an IJEPALatentVisualizationDecoderTrainer object.
+
+        Args:
+            partial_dataloader: A partially instantiated dataloader lacking a provided dataset.
+            partial_optimizer: A partially instantiated optimizer lacking provided parameters.
+            device: The accelerator device (e.g., CPU, GPU) utilized for training the model.
+            logger: The logger object for recording training metrics and visualizations.
+            encoder_name: Name of the encoder (context or target) whose latent space will be visualized.
+            decoder_name: Name of the decoder (context or target) to be trained for visualization.
+            max_epochs: Maximum number of epochs to train the decoder. Default is 1.
+            minimum_dataset_size: Minimum number of samples required in the dataset to start training. Default is 1.
+            minimum_new_data_count: Minimum number of new data samples required to run the training. Default is 0.
+            num_visualize_images: Number of images to use for visualization. Default is 64.
+            visualize_grid_row: Number of images per row in the visualization grid. Default is 8.
+        """
+        super().__init__()
+
+        self.partial_optimizer = partial_optimizer
+        self.partial_dataloader = partial_dataloader
+        self.device = device
+        self.logger = logger
+        self.log_prefix = "i-jepa-latent-visualization/"
+
+        # Checks the decoder name correspond to encoder name.
+        match encoder_name:
+            case ModelNames.I_JEPA_CONTEXT_ENCODER:
+                assert (
+                    decoder_name == ModelNames.I_JEPA_CONTEXT_DECODER
+                ), "Context encoder must be paired with context decoder"
+                self.log_prefix += "context/"
+            case ModelNames.I_JEPA_TARGET_ENCODER:
+                assert (
+                    decoder_name == ModelNames.I_JEPA_TARGET_DECODER
+                ), "Target encoder must be paired with target decoder"
+                self.log_prefix += "target/"
```

Comment:
上の`__init__`で`decoder_name`だけを受け取るようにすれば、例えばこの箇所はこんな感じに簡素にできる
```suggestion
        # Prepare self.log_prefix correspond to decoder name.
        match decoder_name:
            case ModelNames.I_JEPA_CONTEXT_DECODER:
                encoder_name = ModelNames.I_JEPA_CONTEXT_ENCODER
                self.log_prefix += "context/"
            case ModelNames.I_JEPA_TARGET_DECODER:
                encoder_name = ModelNames.I_JEPA_TARGET_ENCODER
                self.log_prefix += "target/"
```